File: patch_cache.js
const fs = require('fs');

function updateCache(filepath) {
  let content = fs.readFileSync(filepath, 'utf8');
  content = content.replace(/\?v=(\d+)/g, (match, p1) => {
    return `?v=${parseInt(p1) + 1}`;
  });
  fs.writeFileSync(filepath, content);
  console.log('Updated cache version in ' + filepath);
}

updateCache('public/viewer.html');
updateCache('public/admin/tagger.html');

File: patch_cache.py
import re

with open('public/viewer.html', 'r') as f:
    content = f.read()

# Update script cache busts in viewer.html
pattern = r"""\?v=1\.2\.1"""
replacement = r"""?v=1.2.2"""

content = re.sub(pattern, replacement, content)

with open('public/viewer.html', 'w') as f:
    f.write(content)
